src/smm/sm-cwa14890.c

```diff
@@ -139,6 +139,8 @@ sm_cwa_decode_authentication_data(struct sc_context *ctx, struct sm_cwa_keyset *
 		LOG_FUNC_RETURN(ctx, SC_ERROR_SM_AUTHENTICATION_FAILED);
 
 	rv = sm_decrypt_des_cbc3(ctx, keyset->enc, session_data->mdata, session_data->mdata_len, &decrypted, &decrypted_len);
+	if (rv != SC_SUCCESS)
+		free(decrypted);
```

Comment:
would not be better if the `sm_decrypt_des_cbc3()` would not allocate/free the output parameter if it fails? With this you will have to fix the same issue in `sm-card-iasecc.c`.

Reply:
I didn't want to refactor this too much, but I can try to.

Reply:
done